### DISCUSSION TITLE
how to get self.paths.data from  modules not "self" aware?

### DISCUSSION THREAD

**Discussion by l-o-u-i-s-g** [beeware/toga#2612] [Q&A] [answered]
I'm trying to segregate my  sqlite code from the UI code.  
I can get the path for my sqlite database file  with self.paths.data  in the startup function but then how can I share that information with my other modules without having to pass self down the tree? 
Is there a way to set application wide global variables at run time?  That I could then call from my   my_db.py file for example?
or should I use environment variables? 
any other ideas on how would be the clean way to do this?

**Comment by rmartin16** [comment, marked as answer]
Since Toga only allows a single app to run in a process, you can access the app via `toga.App.app`. You'll just need to make sure your `toga.App` has been instantiated first.
